### PR TITLE
GUI: Convert loading screen's progress bar to image

### PR DESCRIPTION
**Pull request by kilbith**
This allows to have custom progress bars with images. ~Image scaling still need to be implemented.~

loadscreen

### CODE REVIEW

**Comment by Zeno-** [comment]
looks very nice!

File: src/drawscene.cpp [hunk 1]
```diff
@@ -592,28 +593,28 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
...
-				barrect.LowerRightCorner-1), NULL); // black inside the bar
-		driver->draw2DRectangle(video::SColor(255, 128, 128, 128), core::rect<s32> (
-				barrect.UpperLeftCorner + 1,
-				core::vector2d<s32>(
-						barrect.LowerRightCorner.X -
-						(barsize.X - 1) + percent * (barsize.X - 2) / 100,
-						barrect.LowerRightCorner.Y - 1)), NULL); // the actual progress
+		std::string gamepath = fs::RemoveRelativePathComponents(
+			porting::path_share + DIR_DELIM + "textures");
+		video::ITexture *progress_img = driver->getTexture(
```
Comment (Zeno-) [inline src/drawscene.cpp]: Why is DIR_DELIM used above and not here?

File: src/drawscene.cpp [hunk 2]
```diff
@@ -592,28 +593,28 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
...
+			(gamepath + "/base/pack/progress_bar.png").c_str());
+		video::ITexture *progress_img_bg = driver->getTexture(
+			(gamepath + "/base/pack/progress_bar_bg.png").c_str());
+
+		const core::dimension2d<u32> img_size = progress_img_bg->getSize();
```
Comment (nerzhul) [inline src/drawscene.cpp]: const ref instead of const

**Review by nerzhul — CHANGES_REQUESTED** [review]
except the little thing okay for me

**Comment by kilbith** [comment]
Updated with image scaling and hopefully ready for merge.

File: src/drawscene.cpp [hunk 3]
```diff
@@ -592,28 +593,35 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
...
+			porting::path_share + DIR_DELIM + "textures");
+		video::ITexture *progress_img = driver->getTexture(
+			(gamepath + "/base/pack/progress_bar.png").c_str());
+		video::ITexture *progress_img_bg = driver->getTexture(
+			(gamepath + "/base/pack/progress_bar_bg.png").c_str());
+
+		const core::dimension2d<u32> &img_size = progress_img_bg->getSize();
+		u32 imgW = MYMAX(208, img_size.Width);
+		u32 imgH = MYMAX(24, img_size.Height);
+		v2s32 img_pos((screensize.X / 2) - imgW / 2,
```
Comment (nerzhul) [inline src/drawscene.cpp]: please factorize division operations mathematicly

File: src/drawscene.cpp [hunk 4]
```diff
@@ -592,28 +593,35 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
 	// draw progress bar
 	if ((percent >= 0) && (percent <= 100))
 	{
```
Comment (nerzhul) [inline src/drawscene.cpp]: can you fix this brace please ?

**Review by nerzhul — CHANGES_REQUESTED** [review]

**Review by nerzhul — APPROVED** [review]

**Review by sfan5 — APPROVED** [review]

**Comment by tobyplowy** [comment]
@kilbith +1

**Comment by est31** [comment]
Nice change!

Who should be attributed as author of the textures in README?

**Comment by kilbith** [comment]
@est31 Me, I made them in hurry. We'll have different textures in our engine.

**Comment by NathanSalapat** [comment]
I assume this will not be possible to use on a server, as the server wouldn't send the image before anything starts loading?

**Comment by sofar** [comment]
@NathanSalapat I assume not.

Can the images from texture packs override the default ones?

Can the images from subgames (for singleplayer games) override the default ones?

**Comment by kilbith** [comment]
https://github.com/minetest/minetest/pull/5368

**Pull request merged**